### ISSUE TITLE
Opening advanced inventory with an empty dragged vehicle selected and on top of items breaks the "dragged" pointer

### ISSUE THREAD

**Issue by zachary-kaelan** [CleverRaven/Cataclysm-DDA#48469]
# Describe the bug

if you try to use advanced inventory with an empty, dragged vehicle that is on top of items, it will cause `[D]` to point to the tile beneath the vehicle. This actually persists even if you move the vehicle off of the items, causing everything you try to move into the vehicle using advanced inventory to instead place it onto whatever tile the vehicle's cargo space is currently on. To fix it, you have to change both sides to be pointed somewhere other than `[D]` and then change it back, which works even if the vehicle is still on top of the items. Placing an item into the vehicle using `D`rop will also fix it.

# Steps To Reproduce

1. Spawn a shopping cart, empty it, and start dragging it.
2. Open advanced inventory and set one side to point to `[D]`.
3. Close advanced inventory and position the vehicle on top of items.
4. Re-open advanced inventory and it will show the items that are on the floor below it.
5. Change it to point away from `[D]` then change back, and it will no longer show the items.

# Expected behavior

It is expected for items moved into `[D]` using advanced inventory to move into the dragged vehicle and not the floor. I often will not notice when `[D]` has broken and notice that my cart is empty and all the items were just moved from the cabinet to the floor. 

It could be especially disastrous if attempting to move unpulped zombie corpses from other storage into the vehicle (saving them to dissect later), as it would place them on the floor instead and they would immediately resurrect. 

# Screenshots

![image](https://user-images.githubusercontent.com/30245333/114279294-d68ad680-9a01-11eb-9257-c60bf778006d.png)
![image](https://user-images.githubusercontent.com/30245333/114279303-e4d8f280-9a01-11eb-8264-0917c164bb04.png)
![image](https://user-images.githubusercontent.com/30245333/114279310-eefaf100-9a01-11eb-9ec3-e7810a0a8224.png)
![image](https://user-images.githubusercontent.com/30245333/114279318-fc17e000-9a01-11eb-86c1-f69f929c7463.png)

# Versions and configuration

- OS: Windows 7 Professional
    - OS Version: 6.1.7601 Service Pack 1 Build 7601
- Game Version: 0.E-10361-gfa0b7d6 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food]
]

**Comment by OliveChukar** [comment]
Tested on 0.E-10383-gad91f6a [64-bit] with  Dark Days Ahead [dda] only no mods.  Was able to replicate. Dragging a shopping cart over items with advanced inventory set to (D) showed items in the cart when they were really on the ground if the cart was empty. If there were items in the cart it displayed correctly.  After getting the display issue I set advanced inventory to the space with the items then back to (D) this corrected the display but broke advanced inventory so that items were placed on the ground rather then in the cart when moving them to (D).
